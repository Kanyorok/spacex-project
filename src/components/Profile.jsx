import { useSelector } from 'react-redux';

const Profile = () => {
  const { missions } = useSelector((state) => state.missions);
  const missionsJoined = missions.filter((mission) => mission.reserved);
  return (
    <div className="max-w-6xl mx-auto">
      <div className="flex gap-5">
        <div>
          <h1 className="font-bold">Missions</h1>
          <table className="table-fixed border dark:border-neutral-500 mt-4 mb-4">
            <tbody>
              {missionsJoined.map((mission) => (
                <tr key={mission.mission_id}>
                  <td className="w-3/6 whitespace-wrap border-b border-r px-6 py-4 dark:border-neutral-500">
                    {mission.mission_name}
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>

      </div>
    </div>
  );
};
export default Profile;
