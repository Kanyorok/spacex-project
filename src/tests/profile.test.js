import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Profile from '../components/Profile';

const mockStore = configureStore([]);
const initialState = {
  missions: {
    missions: [
      {
        mission_id: '1',
        mission_name: 'Mock Mission 1',
        description: 'The best mission of space',
        reserved: true,
      },
    ],
  },
  rockets: {
    rockets: [
      {
        id: '1',
        name: 'Falcon 2',
        description: 'It is a good flight to the space',
        reserved: false,
      },
    ],
  },
};
const store = mockStore(initialState);
test('The profile component renders i a good way', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Profile />
      </BrowserRouter>
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
