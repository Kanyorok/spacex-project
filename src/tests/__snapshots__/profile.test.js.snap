// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The profile component renders i a good way 1`] = `
<DocumentFragment>
  <div
    class="max-w-6xl mx-auto"
  >
    <div
      class="flex gap-5"
    >
      <div>
        <h1
          class="font-bold"
        >
          Missions
        </h1>
        <table
          class="table-fixed border dark:border-neutral-500 mt-4 mb-4"
        >
          <tbody>
            <tr>
              <td
                class="w-3/6 whitespace-wrap border-b border-r px-6 py-4 dark:border-neutral-500"
              >
                Mock Mission 1
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div>
        <h1
          class="font-bold"
        >
          Rockets
        </h1>
        <table
          class="table-fixed border dark:border-neutral-500 mt-4 mb-4"
        >
          <tbody />
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
